swupd/create_manifests.go

```diff
@@ -270,24 +271,64 @@ func processBundles(ui UpdateInfo, c config, numWorkers int) ([]*Manifest, error
 
 func addUnchangedManifests(appendTo *Manifest, appendFrom *Manifest, bundles []string) {
 	for _, f := range appendFrom.Files {
-		if f.findFileNameInSlice(appendTo.Files) != nil {
-			continue
-		}
-
 		if f.Name == IndexBundle {
 			// this is generated new each time
 			continue
 		}
 
+		bundleName := f.Name
+		if f.Type == TypeIManifest {
```

Comment:
This can all be rewritten as
```go
if f.findFileNameInSlice(appendTo.Files) != nil {
    continue
}
bundleName := f.Name
if f.Type == TypeIManifest {
    bundleName = strings.Split(bundleName, ".I.")[0]
}
```

Reply:
I just noticed this didn't get simplified as I requested. Can that be done before merge please?

Reply:
@matthewrsj There is an problem with this suggested change:

This change can add stale iterative manifests that are no longer valid. For instance, if the FROM MoM has an entry for bundle1.I.10 and this bundle was updated, then the TO MoM would have entries for bundle1.I.10 and bundle1.I.20. In this case, bundle1.I.10 would no longer be valid and should be omitted from the TO MoM.

Would it be sufficient to add a comment here or do I need to look at refactoring this implementation?

Reply:
I created a generic findBundleInManifest to look for bundle names in both IManifest and Manifests. This function won't work for other files, so I'm returning nil in that case. Not sure if it's really better than use the version before that, but I'd let you decide. If you have any other implementation ideas, we can rewrite that too.

Reply:
You're right, totally thought those `f.funcName1` and `f.funcName2` were the same functions...